app/routes/circulars/circulars.lib.ts

```diff
@@ -76,9 +80,17 @@ const subjectMatchers: SubjectMatcher[] = [
     /Baksan\sNeutrino\sObservatory\sAlert[.\s_-]*(\d{6}.\d{2})/i,
     ([, id]) => `Baksan Neutrino Observatory Alert ${id}`,
   ],
-  [/EP[.\s_-]*(\d{6}[a-z])/i, ([, id]) => `EP${id}`],
-  [/FRB[.\s_-]*(\d{8}[a-z])/i, ([, id]) => `FRB ${id}`.toUpperCase()],
+  [/EP[.\s_-]*(\d{6}[a-z])/i, ([, id]) => `EP${id.toLowerCase()}`],
+  [
+    /SN[.\s_-]*(\d{4}[a-zA-Z]*)/i,
+    ([, id]) => `SN ${id.length == 5 ? id.toUpperCase() : id.toLowerCase()}`,
+  ],
+  [/GW[.\s_-]*(\d{6})/i, ([, id]) => `GW${id}`],
+  [/FRB[.\s_-]*(\d{8}[a-zA-Z])/i, ([, id]) => `FRB ${id}`.toUpperCase()],
+  [/FRB[.\s_-]*(\d{6}[a-zA-Z]?)/i, ([, id]) => `FRB ${id}`.toUpperCase()],
```

Comment:
Please merge this with the above regular expression. You can express the two possible ID formats by combining them with the `|` token.